src/setup_payload/SetupPayload.cpp

```diff
@@ -207,6 +208,33 @@ CHIP_ERROR SetupPayload::removeSerialNumber()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR SetupPayload::generateRandomSetupPin(uint32_t & setupPINCode)
+{
+    uint8_t retries          = 0;
+    const uint8_t maxRetries = 255;
+
+    do
+    {
+        ReturnErrorOnFailure(Crypto::DRBG_get_bytes(reinterpret_cast<uint8_t *>(&setupPINCode), sizeof(setupPINCode)));
+
+        // Passcodes shall be restricted to the values 00000001 to 99999998 in decimal, see 5.1.1.6
+        setupPINCode = (setupPINCode % kSetupPINCodeMaximumValue) + 1;
```

Comment:
Pre-existing, and not trivial to fix, but this does not produce uniform distribution of setup codes....  Maybe worth a TODO comment?  Not sure how important uniformity is here in practice.

Reply:
Yaa, I am not sure why uniformity will be important here. I added a TODO anyways in case we wanted to make it uniform.